:docs: Adding a "what is", file structure info (#333)
terra-project/docs: docs/develop/token-factory/README.md

> # Build a CW20 Tokens Factory  

Author: ```suggestion
# Build a CW20 Token Factory  
```

> This tutorial describes how to build a CW20 Tokens factory. 

Author: ```suggestion
This tutorial describes how to build a CW20 Token Factory. 
```

> To begin, learn what CW20 Tokens factory is,

Author: ```suggestion
To begin, learn what CW20 Token Factory is,
```

and then get started building.
 

> # What is a CW20 Tokens Factory?

Author: ```suggestion
# What is a CW20 Token Factory?
```

> A CW20 Tokens Factory is a Terra DApp that mint, burn, track 

Author: ```suggestion
A CW20 Tokens Factory is a Terra DApp that mints, burns, tracks 
```

typos

> A CW20 Tokens Factory is a Terra DApp that mints, burns, tracks 

Author: ```suggestion
A CW20 Token Factory is a Terra dApp that mints, burns, tracks 
```

> and increase the supply of custom

Author: ```suggestion
and increases the supply of custom
```

typo
